indico/http/client.py

```diff
@@ -164,13 +170,16 @@ def _make_request(
         elif response.status_code == 401 and _refreshed:
             raise IndicoAuthenticationFailed()
 
-        if response.status_code == 503 and 'Retry-After' in response.headers:
-            raise IndicoHibernationError(after=response.headers.get('Retry-After'))
+        if response.status_code == 503 and "Retry-After" in response.headers:
+            raise IndicoHibernationError(after=response.headers.get("Retry-After"))
 
         if response.status_code >= 500:
-            raise IndicoRequestError(
-                code=response.status_code
-            )
+            raise IndicoRequestError(code=response.status_code)
+
+        # Successful signed upload returns no content, so early exit
+        # before we crash trying to deserialize anything
+        if response.status_code == 200 and response.headers.get("Server") == "AmazonS3":
+            return
```

Comment:
I'd say just check for application/json content type in the response headers and just return .text instead if its not json

Reply:
i think we had that catch there for when actually uploading the file to amazon - since when grabbing the signed url , the response object has application/json content type

Reply:
The response from Amazon on the upload will not be applicaiton./json though

Reply:
oh i misinterpreted what you meant by that we want to check that the response content type is not applicaiton/json

Reply:
Mhm, so I'm thinking something that looks like:

```python
   content = await serializers_dict[response.headers.get("content-type").lower()](response)
   return content
```

Reply:
idk what deserializers are supported but yeah..

Reply:
did you want to support all of the serialization methods as defined here? https://github.com/IndicoDataSolutions/indico-client-python/blob/master/indico/http/serialization.py